File: src/client/java/vergisst/minecraftmod/weaponthrow/client/mixins/PlayerEntityRendererMixin.java
package vergisst.minecraftmod.weaponthrow.client.mixins;

import net.minecraft.client.network.AbstractClientPlayerEntity;
import net.minecraft.client.render.entity.PlayerEntityRenderer;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.injection.At;
import org.spongepowered.asm.mixin.injection.Inject;
import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
import vergisst.minecraftmod.weaponthrow.client.events.OnStartPlayerRender;

@Mixin(PlayerEntityRenderer.class)
public class PlayerEntityRendererMixin {
    @Inject(method = "setModelPose", at = @At("TAIL"))
    private void setModelPose(AbstractClientPlayerEntity player, CallbackInfo info) {
        OnStartPlayerRender.EVENT.invoker().interact(
                (PlayerEntityRenderer) (Object) this,
                player
        );
    }
}

File: src/client/kotlin/vergisst/minecraftmod/weaponthrowlite/client/mixins/PlayerEntityRendererMixin.kt
package vergisst.minecraftmod.weaponthrowlite.client.mixins

import net.minecraft.client.network.AbstractClientPlayerEntity
import net.minecraft.client.render.entity.PlayerEntityRenderer
import org.spongepowered.asm.mixin.Mixin
import org.spongepowered.asm.mixin.injection.At
import org.spongepowered.asm.mixin.injection.Inject
import org.spongepowered.asm.mixin.injection.callback.CallbackInfo
import vergisst.minecraftmod.weaponthrowlite.client.events.OnStartPlayerRender

@Mixin(PlayerEntityRenderer::class)
class PlayerEntityRendererMixin {
    @Inject(method = ["setModelPose"], at = [At("TAIL")])
    private fun setModelPose(player: AbstractClientPlayerEntity, info: CallbackInfo?) {
        OnStartPlayerRender.EVENT.invoker().interact((this as Any as PlayerEntityRenderer), player)
    }
}